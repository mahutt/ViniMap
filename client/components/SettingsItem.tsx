import { Link } from 'expo-router';
import React from 'react';
import { StyleSheet, Text, View, Pressable } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

interface SettingsItemProps {
  title: string;
  route: string;
}

const SettingsItem: React.FC<SettingsItemProps> = ({ title, route }) => {
  return (
    <Link href={route as any} asChild>
      <Pressable style={({ pressed }) => [styles.container, pressed && styles.pressed]}>
        <View style={styles.inner}>
          <Text style={styles.title}>{title}</Text>
          <Ionicons name="chevron-forward" size={24} color="#666" />
        </View>
      </Pressable>
    </Link>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    backgroundColor: '#F0F0F0',
    marginBottom: 16,
    borderRadius: 16,
  },
  inner: {
    padding: 20,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  pressed: {
    opacity: 0.7,
  },
  title: {
    fontSize: 17,
    fontWeight: '500',
    color: '#333',
  },
});

export default SettingsItem;
