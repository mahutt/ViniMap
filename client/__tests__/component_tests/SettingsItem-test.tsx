import React from 'react';
import { render } from '@testing-library/react-native';
import { Pressable } from 'react-native';
import SettingsItem from '@/components/SettingsItem';

// Mock the expo-router Link component
jest.mock('expo-router', () => ({
  Link: ({ children, href }: { children: React.ReactNode; href: string }) => children,
}));

// Mock the Ionicons component
jest.mock('@expo/vector-icons', () => ({
  Ionicons: () => 'IconMock',
}));

describe('SettingsItem', () => {
  const props = {
    title: 'Test Title',
    route: '/test-route',
  };

  it('renders correctly with all props', () => {
    const { getByText, toJSON } = render(<SettingsItem {...props} />);

    expect(getByText('Test Title')).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });

  it('handles pressed state styling', () => {
    const { UNSAFE_getByType } = render(<SettingsItem {...props} />);

    const pressable = UNSAFE_getByType(Pressable);

    const regularStyle = pressable.props.style({ pressed: false });
    const pressedStyle = pressable.props.style({ pressed: true });

    expect(regularStyle).not.toEqual(pressedStyle);

    expect(pressedStyle.find((style: any) => style && style.opacity === 0.7)).toBeTruthy();
  });
});
