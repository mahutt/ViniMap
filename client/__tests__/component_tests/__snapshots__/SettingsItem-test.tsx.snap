// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsItem renders correctly with all props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#F0F0F0",
        "borderRadius": 16,
        "marginBottom": 16,
        "width": "100%",
      },
      false,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 17,
          "fontWeight": "500",
        }
      }
    >
      Test Title
    </Text>
    IconMock
  </View>
</View>
`;
